ppa/archive/gale.py

```diff
@@ -218,19 +226,23 @@ def get_item_pages(self, item_id, gale_record=None):
         if gale_record is None:
             gale_record = self.get_item(item_id)
 
+        # todo: needs try/catch error handling
+        local_ocr_text = get_local_ocr(item_id)
+
         # iterate through the pages in the response
         for page in gale_record["pageResponse"]["pages"]:
             page_number = page["pageNumber"]
 
             # Fetch higher quality local OCR text if possible, with fallback to Gale
             # OCR. Set a tag to indicate the usage of local OCR text.
             tags = []
-            try: 
-                ocr_text = get_local_ocr(item_id, page_number)
-                tags = ["local_ocr"]
-            except FileNotFoundError as e:
+            ocr_text = local_ocr_text.get(page_number)
+            # TODO: set local ocr tag here
+            if not ocr_text and page_number not in local_ocr_text:
+                # if ocr text is unset and page number is not present,
+                # try getting the ocr from the gale api result
+                logger.warning(f"No local OCR for {item_id} {page_number}")
                 ocr_text = page.get("ocrText")  # some pages have no text
-                logger.warning(f'Local OCR not found for {item_id} {page_number}')
```

Comment:
Seems like this would be better as an if/else clause, since the only time the Gale text should be fetched is when the page number is not in `local_ocr_text`

Reply:
ah, good point... there are two cases here, aren't there? we could have the page number in our json file and have no text, or it might not be present at all (e.g., we missed it somehow)

Reply:
Leaving this comment to remind you to confirm that I've done what you were suggesting.

I got a code complexity alert for this function - was able to reduce complexity slightly. If you see any easy wins, please suggest them.